### PR TITLE
feat: add poetry pre-commit hooks

### PR DESCRIPTION
**Pull request by JacobCoffee**
### Pull Request Checklist

[//]: # "Please review the [Litestar contribution guidelines](https://github.com/litestar-org/litestar/blob/main/CONTRIBUTING.rst) for this repository."

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR

[//]: # "By submitting this issue, you agree to:"
[//]: # "- follow Litestar's [Code of Conduct](https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md)"
[//]: # "- follow Litestar's [Contribution Guidelines](https://litestar.dev/community/contribution-guide)"

### Description

Adds a pre-commit hook for Poetry, checking the lock file and `pyproject.toml` structure

[//]: # "Please describe your pull request for new release changelog purposes"
[//]: # "Example: 'This pr adds the capability to use server-sent events, and documents it in the usage docs'"

### Close Issue(s)

[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

### CODE REVIEW

**Comment by JacobCoffee** [comment]
```
 poetry-check.............................................................Failed
- hook id: poetry-check
- exit code: 1

Error: Cannot find dependency "tortoise-orm" for extra "tortoise-orm" in main dependencies.
Error: Cannot find dependency "python-dateutill" for extra "full" in main dependencies.
```

._.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -12,6 +12,11 @@ repos:
       - id: end-of-file-fixer
       - id: mixed-line-ending
       - id: trailing-whitespace
+  - repo: https://github.com/python-poetry/poetry
+    rev: "1.5.1"
+    hooks:
+      - id: poetry-check
+      - id: poetry-lock
```
[the remark this answers is not in the document]
Comment (cofin) [reply .pre-commit-config.yaml:19]: Why don't we want this change?

**Review by cofin — APPROVED** [review]
LGTM

**Comment by provinzkraut** [comment]
This seems to have slowed down our pre-commit significantly.
The `poetry-lock` step takes ~15 seconds each run, about 1/3 of the total time, including mypy and pyright.

**Pull request merged**